masp_primitives/src/transaction/components/amount.rs

```diff
@@ -143,180 +176,331 @@ impl Amount<AssetType> {
     }
 }
 
-impl<Unit: Hash + Ord + BorshSerialize + BorshDeserialize> From<Unit> for Amount<Unit> {
+impl<Unit, Magnitude> From<Unit> for Amount<Unit, Magnitude>
+where
+    Unit: Hash + Ord + BorshSerialize + BorshDeserialize,
+    Magnitude: BorshSerialize + BorshDeserialize + PartialEq + Eq + Copy + From<bool>,
```

Comment:
I find the `From<bool>` a bit strange. What about `From<u8>` or `From<u64>`. I'm thinking ahead about putting namada's `Uint` type in here.